[MRG] Convert docstring of Axes.pie() into numpydoc

kahnchana opened the pull request:
#### Reference Issue #8272

#### What does this implement/fix? Explain your changes.
The docstring of the function pie() in lib/matplotlib/axes/_axes.py has been converted to numpydoc format. Also, pep-8 style errors in the code have been fixed.

QuLogic commented:
There's a lot of extraneous whitespace changes here. While correcting this is not unwelcome, it makes it rather difficult to say what parts of this change are actually relevant. It would be better to have a separate PR for whitespace correction, or only do it on the parts that you are actually editing.

Also, please write actual commit message, not just a single dot.

kahnchana commented:
Added the changes you suggested. I removed all the whitespace fixing. Will create a separate issue for that later. And added the exact commit name. Sorry about previous mistake.

QuLogic commented:
Hmm, I should have been a bit clearer. It's not good to revert this because now there's a commit making the whitespace changes and another commit undoing those, and then if you open a separate PR, there'll be another commit fixing the whitespace again.

If you know how to rebase, it would be a little better if you rebased and squashed these commits all into a single one.

QuLogic commented on lib/matplotlib/axes/_axes.py:
```diff
@@ -2475,99 +2475,102 @@ def pie(self, x, explode=None, labels=None, colors=None,
+            The input array used to make the pie chart.
 
-          *explode*: [ *None* | len(x) sequence ]
+        explode: None or array
```
Add a space before the colon for all parameters.

QuLogic commented on lib/matplotlib/axes/_axes.py:
```diff
@@ -2475,99 +2475,102 @@ def pie(self, x, explode=None, labels=None, colors=None,
             Draw a shadow beneath the pie.
 
-          *startangle*: [ *None* | Offset angle ]
+        startangle: None or Offset angle
```
`Offset angle` is not a type; should be `float`.

QuLogic commented on lib/matplotlib/axes/_axes.py:
```diff
@@ -2475,99 +2475,102 @@ def pie(self, x, explode=None, labels=None, colors=None,
-
-          *frame*: [ *False* | *True* ]
-            Plot axes frame with the chart.
+        center:  list of int
```
Extra space. Also, why `int`?

kahnchana replied:
My bad. Changing to float.

QuLogic commented on lib/matplotlib/axes/_axes.py:
```diff
@@ -2475,99 +2475,102 @@ def pie(self, x, explode=None, labels=None, colors=None,
-          ax = axes([0.1, 0.1, 0.8, 0.8])
+        Returns
+        -------
+        patches: list
```
Also add space before colon of every return value.

QuLogic commented on lib/matplotlib/axes/_axes.py:
```diff
@@ -2475,99 +2475,102 @@ def pie(self, x, explode=None, labels=None, colors=None,
-          axes(aspect=1)
+        autotexts: list
+            A is a list of :class:`~matplotlib.text.Text` instances for the
+            numeric labels. Is returned only is if parameter *autopct* is
```
is if -> if

QuLogic commented on lib/matplotlib/axes/_axes.py:
```diff
@@ -2475,99 +2475,102 @@ def pie(self, x, explode=None, labels=None, colors=None,
+            A is a list of :class:`~matplotlib.text.Text` instances for the
+            numeric labels. Is returned only is if parameter *autopct* is
+            not *None*. If *autopct* is *None*, the tuple returned
+            is (patches, texts).
```
Leave out this last sentence.

kahnchana commented:
I tried rebasing and made a mistake. I ended up adding additional commits. I am trying to fix this and figure out the rebasing. 

Also made the changes suggested.

kahnchana commented:
Shall I remove this PR and open a new PR with the changes you suggested?

dstansby commented:
It looks like quite a few things have gone wrong with git here; if it would be easier for you to open a new PR, please feel free!

phobson commented:
@kahnchana I have cleaned up the commits to this branch locally.

I'm closing this in favor of #8371
